[jit] Delete polyfill typing

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#27510 [jit] Delete polyfill typing**

Summary:
We could delete polyfill typing bc requirements.txt require user to
install typing as a dependency whether in py2 or py3, so those typing
actually not getting used either ways.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27510

Differential Revision: [D20673393](https://our.internmc.facebook.com/intern/diff/D20673393)

wanchaol committed:
[jit] Delete polyfill typing

Summary:
We could delete polyfill typing bc requirements.txt require user to
install typing as a dependency whether in py2 or py3, so those typing
actually not getting used either ways.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

eellison commented on torch/_jit_internal.py:
```diff
@@ -485,110 +486,43 @@ def _get_overloaded_methods(method, mod_class):
 
-try:
-    import typing
-    from typing import Tuple, List, Dict, Optional
```
Can you leave the imports in so that any other libraries depending on torch.jit.annotations don't fail now ? These imports are used in torchvision.

wanchaol replied:
sure

wanchaol committed:
Update on "[jit] Delete polyfill typing"


Summary:
We could delete polyfill typing bc requirements.txt require user to
install typing as a dependency whether in py2 or py3, so those typing
actually not getting used either ways.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

wanchaol committed:
rebase on "[jit] Delete polyfill typing"


Summary:
We could delete polyfill typing bc requirements.txt require user to
install typing as a dependency whether in py2 or py3, so those typing
actually not getting used either ways.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27510

wanchaol committed:
Update on "[jit] Delete polyfill typing"


Summary:
We could delete polyfill typing bc requirements.txt require user to
install typing as a dependency whether in py2 or py3, so those typing
actually not getting used either ways.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27510

[ghstack-poisoned]

wanchaol committed:
Update on "[jit] Delete polyfill typing"


Summary:
We could delete polyfill typing bc requirements.txt require user to
install typing as a dependency whether in py2 or py3, so those typing
actually not getting used either ways.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27510

[ghstack-poisoned]

wanchaol committed:
Update on "[jit] Delete polyfill typing"


Summary:
We could delete polyfill typing bc requirements.txt require user to
install typing as a dependency whether in py2 or py3, so those typing
actually not getting used either ways.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27510

[ghstack-poisoned]

wanchaol committed:
Update on "[jit] Delete polyfill typing"


Summary:
We could delete polyfill typing bc requirements.txt require user to
install typing as a dependency whether in py2 or py3, so those typing
actually not getting used either ways.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27510

[ghstack-poisoned]

wanchaol committed:
Update on "[jit] Delete polyfill typing"


Summary:
We could delete polyfill typing bc requirements.txt require user to
install typing as a dependency whether in py2 or py3, so those typing
actually not getting used either ways.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27510

[ghstack-poisoned]

wanchaol committed:
Update on "[jit] Delete polyfill typing"


Summary:
We could delete polyfill typing bc requirements.txt require user to
install typing as a dependency whether in py2 or py3, so those typing
actually not getting used either ways.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27510

[ghstack-poisoned]

wanchaol committed:
Update on "[jit] Delete polyfill typing"


Summary:
We could delete polyfill typing bc requirements.txt require user to
install typing as a dependency whether in py2 or py3, so those typing
actually not getting used either ways.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27510

Differential Revision: [D20673393](https://our.internmc.facebook.com/intern/diff/D20673393)

[ghstack-poisoned]

wanchaol committed:
Update on "[jit] Delete polyfill typing"


Summary:
We could delete polyfill typing bc requirements.txt require user to
install typing as a dependency whether in py2 or py3, so those typing
actually not getting used either ways.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27510

Differential Revision: [D20673393](https://our.internmc.facebook.com/intern/diff/D20673393)

[ghstack-poisoned]

wanchaol committed:
Update on "[jit] Delete polyfill typing"


Summary:
We could delete polyfill typing bc requirements.txt require user to
install typing as a dependency whether in py2 or py3, so those typing
actually not getting used either ways.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27510

Differential Revision: [D20673393](https://our.internmc.facebook.com/intern/diff/D20673393)

[ghstack-poisoned]
